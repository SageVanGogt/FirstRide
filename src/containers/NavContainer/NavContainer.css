.nav-bar {
  display: block;
  z-index: 1;
  width: 100vw;
  background-color: #404040;
}

.site-name {
  float: left;
  color: white;
  margin-left: 60px;
  font-size: 3rem;
  font-family: 'Roboto', sans-serif;
}

.links {
  float: right;
  margin-top: 55px;
  margin-right: 100px;
}

.nav-link {
  padding: 0;
  border-bottom: 3px solid transparent;
  margin-right: 40px;
  text-decoration: none;
  font-family: 'Lato', sans-serif;
  color: white;
}

.nav-link:hover {
  border-bottom: 3px solid white;  
}

/* .links {
  left: 400px;
} */