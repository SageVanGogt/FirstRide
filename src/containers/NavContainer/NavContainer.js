import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import * as actions from './../../actions/user';
import { removeDestination } from './../../actions/destination';
import { removeCurrentLocation } from './../../actions/currentLocation';
import { removeUserRides } from './../../actions/userRides';
import { removeRides } from './../../actions/rides';
import { removePickups } from './../../actions/pickups';
import './NavContainer.css';

export class NavContainer extends Component {
  constructor(props) {
    super(props);
  }

  handleSignout = () => {
    this.props.logoutUser();
    this.props.toggleLogin();
    this.props.removeUserRides();
  };

  handleReturnHome = () => {
    this.props.removeCurrentLocation();
    this.props.removeDestination();
    this.props.removeRides();
    this.props.removePickups();
  }

  render() {
    return (
      <div className="nav-bar">
        <h1 className="site-name">FirstRide.</h1>
        <div className="links">
          <NavLink 
            className="nav-link" 
            to="/"
            onClick={this.handleReturnHome}>
            Home
          </NavLink>
          <button 
            className="nav-btn"
            onClick={this.props.toggleShowUserRides}>
            My Rides
          </button>
          <NavLink className="nav-link" to="/profile">Me</NavLink>
          { 
            this.props.user.id ? 
              <button 
                className="nav-btn"
                onClick={this.handleSignout}>
                Signout
              </button> :
              <button 
                className="nav-btn signin-btn" 
                onClick={this.props.toggleLogin}>
                Signin / Signup
              </button>
          }
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
});

export const mapDispatchToProps = (dispatch) => ({
  logoutUser: () => dispatch(actions.signoutUser()),
  removeUserRides: () => dispatch(removeUserRides()),
  removeCurrentLocation: () => dispatch(removeCurrentLocation()),
  removeDestination: () => dispatch(removeDestination()),
  removeRides: () => dispatch(removeRides()),
  removePickups: () => dispatch(removePickups())
});

NavContainer.propTypes = {
  toggleLogin: PropTypes.func,
  toggleShowUserRides: PropTypes.func,
  logoutUser: PropTypes.func,
  removeUserRides: PropTypes.func,
  removeCurrentLocation: PropTypes.func,
  removeDestination: PropTypes.func,
  removeRides: PropTypes.func,
  removePickups: PropTypes.func,
  user: PropTypes.object
};

export default connect(mapStateToProps, mapDispatchToProps)(NavContainer);