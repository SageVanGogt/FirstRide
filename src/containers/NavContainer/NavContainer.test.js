import React from 'react';
import { shallow } from 'enzyme';
import { NavContainer, mapStateToProps } from './NavContainer';

describe('NavContainer', () => {
  let wrapper;
  let mockState;

  beforeEach(() => {
    mockState = {
      user: 'sage',
      rides: [{}, {}]
    }
    wrapper = shallow(<NavContainer 
    {...mockState}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    it('should get the user from the store', () => {
      let expected = 'sage';
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.user;
      
      expect(actual).toEqual(expected);
    })
  })
})