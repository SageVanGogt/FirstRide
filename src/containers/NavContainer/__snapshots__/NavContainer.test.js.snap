// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavContainer
    rides={
        Array [
            Object {},
            Object {},
          ]
    }
    user="sage"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="site-name"
>
          FirstRide
</h1>,
        <div
          className="links"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-link"
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-link"
                    to="/login"
          >
                    Signin
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-link"
                    to="/profile"
          >
                    Me
          </NavLink>
</div>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FirstRide",
          "className": "site-name",
        },
        "ref": null,
        "rendered": "FirstRide",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-link"
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-link"
              to="/login"
>
              Signin
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-link"
              to="/profile"
>
              Me
</NavLink>,
          ],
          "className": "links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "nav-link",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Signin",
              "className": "nav-link",
              "to": "/login",
            },
            "ref": null,
            "rendered": "Signin",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Me",
              "className": "nav-link",
              "to": "/profile",
            },
            "ref": null,
            "rendered": "Me",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="site-name"
>
            FirstRide
</h1>,
          <div
            className="links"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-link"
                        to="/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-link"
                        to="/login"
            >
                        Signin
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-link"
                        to="/profile"
            >
                        Me
            </NavLink>
</div>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FirstRide",
            "className": "site-name",
          },
          "ref": null,
          "rendered": "FirstRide",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-link"
                to="/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-link"
                to="/login"
>
                Signin
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-link"
                to="/profile"
>
                Me
</NavLink>,
            ],
            "className": "links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "className": "nav-link",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Signin",
                "className": "nav-link",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Signin",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Me",
                "className": "nav-link",
                "to": "/profile",
              },
              "ref": null,
              "rendered": "Me",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
